### ISSUE TITLE
Call to undefined method OC\\Server::getEventDispatcher()

### ISSUE THREAD

**Issue by ChristophWurst** [nextcloud/social#1864]
**Describe the bug**

```json
{
  "reqId": "2PtOBQe5Y6B2la734eY8",
  "level": 4,
  "time": "2023-11-06T08:21:35+00:00",
  "remoteAddr": "2001:9e8:367e:ac00:93ff:25b4:2b5:8bf1",
  "user": "--",
  "app": "social",
  "method": "GET",
  "url": "/ocs/v2.php/apps/dashboard/api/v2/widget-items?widgets%5B%5D=activity",
  "message": "Error during app service registration: Call to undefined method OC\\Server::getEventDispatcher()",
  "userAgent": "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/119.0",
  "version": "28.0.0.3",
  "exception": {
    "Exception": "Error",
    "Message": "Call to undefined method OC\\Server::getEventDispatcher()",
    "Code": 0,
    "Trace": [
      {
        "file": "/var/www/nextcloud/apps/social/lib/AppInfo/Application.php",
        "line": 71,
        "function": "registerDeprecatedListener",
        "class": "OCA\\Social\\AppInfo\\Application",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/lib/private/AppFramework/Bootstrap/Coordinator.php",
        "line": 142,
        "function": "register",
        "class": "OCA\\Social\\AppInfo\\Application",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/lib/private/AppFramework/Bootstrap/Coordinator.php",
        "line": 90,
        "function": "registerApps",
        "class": "OC\\AppFramework\\Bootstrap\\Coordinator",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/lib/base.php",
        "line": 700,
        "function": "runInitialRegistration",
        "class": "OC\\AppFramework\\Bootstrap\\Coordinator",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/lib/base.php",
        "line": 1200,
        "function": "init",
        "class": "OC",
        "type": "::"
      },
      {
        "file": "/var/www/nextcloud/ocs/v1.php",
        "line": 31,
        "args": [
          "/var/www/nextcloud/lib/base.php"
        ],
        "function": "require_once"
      },
      {
        "file": "/var/www/nextcloud/ocs/v2.php",
        "line": 23,
        "args": [
          "/var/www/nextcloud/ocs/v1.php"
        ],
        "function": "require_once"
      }
    ],
    "File": "/var/www/nextcloud/apps/social/lib/AppInfo/Application.php",
    "Line": 82,
    "message": "Error during app service registration: Call to undefined method OC\\Server::getEventDispatcher()",
    "exception": {},
    "CustomMessage": "Error during app service registration: Call to undefined method OC\\Server::getEventDispatcher()"
  }
}
```

**To Reproduce**
1. ???

**Expected behavior**

App does not use private APIs.

**Screenshots**

**Client details:**
 - Device: [e.g. iPhone6, desktop]

<details>
<summary>Server details</summary>

**Social app version:**

**Operating system**:

**Web server:**

**Database:**

**PHP version:**

**Nextcloud version:** 28

</details>

<details>
<summary>Logs</summary>

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```

#### Browser log
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```

</details>

**Comment by blindzero** [comment]
same here...
With
**OS:** Debian 6.1.27-1
**PHP:** 8.2.14
**Webserver:** Apache/2.4.57 +  nginx/1.24.0 in proxy-mode (standard Plesk setup)

all clients behave in the same way.

**Comment by sebaxakerhtc** [comment]
Same here.
Steps to reproduce:
At the morning updated to nextcloud 28.0.1
Now installed Nextcloud app for Windows
Runned sync (virtual files)
Recieved 1247 errors :) (Maybe amount of files?)

**Comment by Wunderharke** [comment]
bump.
Same here. Insane log spamming.

**Comment by rikmeijer** [comment]
Fixed it by replacing line ./apps/social/lib/AppInfo/Application.php@71 with: 

$dispatcher = $this->getContainer()->query(IEventDispatcher::class);

and adding (at the top of the file)

use OCP\EventDispatcher\IEventDispatcher;

**Comment by Thatoo** [comment]
@rikmeijer did you make a PR for that?

**Comment by Thatoo** [comment]
Could you write which line you modify and replace by `$dispatcher = $this->getContainer()->query(IEventDispatcher::class); ?`
 because line 71, I have
`71                 $this->registerDeprecatedListener();`
and it doesn't feel right to make the modification there. If you had said to change line 82, I'd understand better : 
`82                 $dispatcher = \OC::$server->getEventDispatcher();`

**Comment by rikmeijer** [comment]
I put my altered Application.php in a [gist](https://gist.github.com/rikmeijer/8b20361f7d2689c663f8e68b50f9133f)

**Comment by paulcalabro** [comment]
Thanks for sharing that @rikmeijer !

**Comment by paulcalabro** [comment]
Is anyone working on this? If not, I can make a PR.

**Comment by teledyn** [comment]
So far as I can tell, Social is abandoned, but you could fork it and merge back if the maintainer ever returns. You Application.php solved the errors for me, and because it was a Vue Event thing I'd hoped it would also solve why I can't any open profiles, but that adventure continues. If you fork this think, I'll subscribe! Because, back when it worked, it was kinda fun.

**Comment by noci2012** [comment]
```
{
  "reqId": "IhilB622InmPCwv23SD0",
  "level": 4,
  "time": "2024-09-24T13:07:15+00:00",
  "remoteAddr": "192.168.7.117",
  "user": "--",
  "app": "social",
  "method": "GET",
  "url": "/apps/logreader/api/poll?lastReqId=RkpiDP9XvTgAf96Sv9l7",
  "message": "Error during app service registration: Call to undefined method OC\\Server::getEventDispatcher()",
  "userAgent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36",
  "version": "30.0.0.14",
  "exception": {
    "Exception": "Error",
    "Message": "Call to undefined method OC\\Server::getEventDispatcher()",
    "Code": 0,
    "Trace": [
      {
        "file": "/var/www/nextcloud/apps/social/lib/AppInfo/Application.php",
        "line": 71,
        "function": "registerDeprecatedListener",
        "class": "OCA\\Social\\AppInfo\\Application",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/lib/private/AppFramework/Bootstrap/Coordinator.php",
        "line": 99,
        "function": "register",
        "class": "OCA\\Social\\AppInfo\\Application",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/lib/private/AppFramework/Bootstrap/Coordinator.php",
        "line": 48,
        "function": "registerApps",
        "class": "OC\\AppFramework\\Bootstrap\\Coordinator",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/lib/base.php",
        "line": 657,
        "function": "runInitialRegistration",
        "class": "OC\\AppFramework\\Bootstrap\\Coordinator",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/lib/base.php",
        "line": 1132,
        "function": "init",
        "class": "OC",
        "type": "::"
      },
      {
        "file": "/var/www/nextcloud/index.php",
        "line": 22,
        "args": [
          "/var/www/nextcloud/lib/base.php"
        ],
        "function": "require_once"
      }
    ],
    "File": "/var/www/nextcloud/apps/social/lib/AppInfo/Application.php",
    "Line": 82,
    "message": "Error during app service registration: Call to undefined method OC\\Server::getEventDispatcher()",
    "exception": [],
    "CustomMessage": "Error during app service registration: Call to undefined method OC\\Server::getEventDispatcher()"
  },
  "id": "66f2b98370f2a"
}
```

**Comment by joshtrichards** [comment]
Fixed in a99b35f7453b72738122f0a6f971f111bc1be6f9
